dbcontrollers/products_test.go

```diff
@@ -269,19 +276,12 @@ func TestCreateProduct(t *testing.T) {
 					return "testPath"
 				})
 
-			if output != nil {
-				if output.Name != expectedData.Name || output.Public != expectedData.Public {
-					t.Errorf(test.TestResultString, testCaseString, output, expectedData)
-					return
-				}
-			} else {
-				if output != expectedData {
-					t.Errorf(test.TestResultString, testCaseString, output, expectedData)
-					return
-				}
+			if diff := pretty.Diff(expectedData, output); len(diff) != 0 {
+				t.Errorf(test.TestResultString, testCaseString, output, expectedData)
```

Comment:
Just an additional option: you could print out the diff to see what happened.    
```suggestion
				t.Errorf(test.TestResultStringWithDiff, testCaseString, output, expectedData, diff)
```